From a3e684353ea351e724ab183bc1ec607f5fc7d837 Mon Sep 17 00:00:00 2001
From: Alexi Demers <alexidemers@gmail.com>
Date: Sat, 22 Apr 2023 20:38:10 -0400
Subject: [PATCH] rockchip-common: configure boot commands for dual bank and
 apparmor=0

apparmor is not enabled, and it must be explicited to avoid k3s from
trying to use it.

The double banking method allows to change the software bank with files
located in /boot, i.e. banka and banb files.
---
 include/configs/rk3328_common.h   |  2 ++
 include/configs/rockchip-common.h | 22 +++++++++++++---------
 2 files changed, 15 insertions(+), 9 deletions(-)

diff --git a/include/configs/rk3328_common.h b/include/configs/rk3328_common.h
index f43fde4..245e657 100644
--- a/include/configs/rk3328_common.h
+++ b/include/configs/rk3328_common.h
@@ -62,6 +62,8 @@
 	RKIMG_DET_BOOTDEV \
 	"partitions=" PARTS_DEFAULT \
 	ROCKCHIP_DEVICE_SETTINGS \
+	BOOTARGS_ROOTA \
+	BOOTARGS_ROOTB \
 	BOOTENV
 
 #endif
diff --git a/include/configs/rockchip-common.h b/include/configs/rockchip-common.h
index e5b8368..724b703 100644
--- a/include/configs/rockchip-common.h
+++ b/include/configs/rockchip-common.h
@@ -135,15 +135,19 @@
 		"setenv devtype ramdisk; setenv devnum 0;" \
 	"fi; \0"
 
-#if defined(CONFIG_AVB_VBMETA_PUBLIC_KEY_VALIDATE)
-#define RKIMG_BOOTCOMMAND			\
-	"boot_android ${devtype} ${devnum};"
-#else
-#define RKIMG_BOOTCOMMAND			\
-	"boot_android ${devtype} ${devnum};"	\
-	"bootrkp;"				\
-	"run distro_bootcmd;"
-#endif
+#define RKIMG_BOOTCOMMAND		\
+	"setenv bootargs \"${bootargs} apparmor=0 \"; \
+	if load mmc 1:3 0x4000000 bankb; then \
+		setenv bootargs \"${bootargs} root=${bootarg_rootb}\"; \
+	else \
+		setenv bootargs \"${bootargs} root=${bootarg_roota}\";\
+	fi; \
+	ext2load mmc 1:3 0x00280000 Image-initramfs-orangepi-r1plus.bin; \
+	ext2load mmc 1:3 0x08300000 rk3328-orangepi-r1-plus.dtb; \
+	booti 0x00280000 - 0x08300000;"
+
+#define BOOTARGS_ROOTA "bootarg_roota=/dev/mmcblk0p4\0"
+#define BOOTARGS_ROOTB "bootarg_rootb=/dev/mmcblk0p5\0"
 
 #endif /* CONFIG_SPL_BUILD */
 
-- 
2.34.1

